### ISSUE TITLE
Secret is unrequired.

### ISSUE THREAD

**Issue by hezedu** [expressjs/session#721]
Session secret is usually saved in clear text. When a hacker can obtain session storage in the server, it is not far from knowing the session secret.

What about use the hash of session.ID as the key store?

**Comment by dougwilson** [comment]
The secret does not encrypt data, so access to the session store will compromise all data, secret or not. The secret is simply to prevent the client who cannot access the server from tampering with the session ID.

If you want to actually encrypt the data saved to your store, you would need to use a store module (or open an issue on one) that supports encrypting the stored data.

**Comment by hezedu** [comment]
Access to the session store will compromise all data, But without secret, you can't get the full id in the cookie. So I think the secret is to prevent the server, not the client.

**Comment by dougwilson** [comment]
I'm not sure what you mean. The full session ID is in the cookie. You don't need the secret to see it.

**Comment by hezedu** [comment]
I mean: When you know session ID and secret on the server, you can get the cookie's ’full session ID.

**Comment by dougwilson** [comment]
That's correct.

**Comment by hezedu** [comment]
But If i use the hash of session.ID as the key store, the Secret looks unnecessary.

**Comment by hezedu** [comment]
If I use the hash of session.id as the key store, even if you know the session ID on the server, you cannot get the full session ID of the cookie. This is what I want to achieve.

**Comment by kohtala** [comment]
@hezehu, you are not being clear on what you are asking. I can not understand you either.

I came on this issue because I am checking what is the reason for the property `secret` on the express-session options object to be required. After some research, I found there is a long discussion on it in #176. For what I understood, it is there due to doubts if the session ID generation is secure enough. I also found https://github.com/nodejs/node/issues/5798 discussing improving it. It may be secure enough, but there was no conclusion if it can be relied to be so always. I also found an excuse that the session ID could be user generated weak id, but fail to see how that would means a secret is always required.

I mention it here to help anyone else looking for same answer, but I can not say if your question is related or not.
